drivers/flash/flash_stm32l5x.c

```diff
@@ -367,10 +367,16 @@ int flash_stm32_write_range(const struct device *dev, unsigned int offset,
 	}
 
 	if (icache_enabled) {
+		int rc2;
+
 		/* Since i-cache was disabled, this would start the
 		 * invalidation procedure, so wait for completion.
 		 */
-		rc = icache_wait_for_invalidate_complete();
+		rc2 = icache_wait_for_invalidate_complete();
+
+		if (!rc) {
+			rc = rc2;
+		}
```

Comment:
```suggestion
                if (rc2 < 0) {
                        LOG_ERR("icache_wait_for_invalidate_complete failed with err %d", rc2);
		        if (!rc) {
			        rc = rc2;
			}
		}
```

Reply:
`icache_wait_for_invalidate_complete` function already has error logging. Is it needed in the caller also?

Reply:
you are right, not needed.